src/register/Register.h

```diff
@@ -0,0 +1,69 @@
+#ifndef WARDUINO_REGISTER_H
+#define WARDUINO_REGISTER_H
+
+#include <Arduino.h>
+#include <Wire.h>
+
+namespace wlp {
+
+    typedef uint16_t size_type;
+
+    class Register {
+        uint8_t m_i2c_addr;
```

Comment:
I think you should make the Register more generic so like it should not be `i2c_adress` but rather the address of any register. If you wanna make something for I2C we should name it I2C or something so it is more clear

Reply:
aight I'll rename it, (I think) it's i2c specfic.